Common++/header/IpAddresses.h

```diff
@@ -0,0 +1,551 @@
+		 * Another example: if subnet is 10.1.X.X, subnet mask is 255.0.0.0 and address is 11.1.1.9 then the method will return false
+		 * @param[in] subnet The subnet to be verified. Notice it's an IPv4Address type, so subnets with don't-cares (like 10.0.0.X) must have some number
+		 * (it'll be ignored if subnet mask is correct)
+		 * @param[in] subnetMask A string representing the subnet mask to compare the address with the subnet
+		 */
+		#if __cplusplus > 201402L || _MSC_VER >= 1900
+		[[deprecated("This method is deprecated and will be deleted in the future. Prefer one of the functions matchSubnet")]]
+		#endif
+		bool matchSubnet(const IPv4Address& subnet, const std::string& subnetMask) const;
+
+		/**
+		 * Checks whether the address matches a subnet.
+		 * Notice this method is deprecated and will be deleted in the future. Prefer one of the functions matchSubnet
+		 * For example: if subnet is 10.1.1.X, subnet mask is 255.255.255.0 and address is 10.1.1.9 then the method will return true
+		 * Another example: if subnet is 10.1.X.X, subnet mask is 255.0.0.0 and address is 11.1.1.9 then the method will return false
+		 * @param[in] subnet The subnet to be verified. Notice it's an IPv4Address type, so subnets with don't-cares (like 10.0.0.X) must have some number
+		 * (it'll be ignored if subnet mask is correct)
+		 * @param[in] subnetMask The subnet mask to compare the address with the subnet
+		 */
+		#if __cplusplus > 201402L || _MSC_VER >= 1900
+		[[deprecated("This method is deprecated and will be deleted in the future. Prefer one of the functions matchSubnet")]]
+		#endif
+		bool matchSubnet(const IPv4Address& subnet, const IPv4Address& subnetMask) const;
+
+		/**
+		 * A static value representing a zero value of IPv4 address, meaning address of value "0.0.0.0"
+		 * Notice this value can be omitted in the user code because the default constructor creates an instance with an unspecified/zero address.
+		 * In order to check whether the address is zero the method isUnspecified can be used
+		 */
+		static const IPv4Address Zero;
+
+	private:
+		uint8_t m_Bytes[4];
+	}; // class IPv4Address
+
+
+	// Implementation of inline methods
+	uint32_t IPv4Address::toUInt() const
+	{
+		uint32_t addr;
+		memcpy(&addr, m_Bytes, sizeof(m_Bytes));
+		return addr;
+	}
+
+
+	// Creation of IPv4Address
+
+	/**
+	 * A function that creates an instance of the class IPv4Address out of 4-byte array.
+	 * @param[in] bytes The address as 4-byte array in network byte order
+	 * @return An instance of class IPv4Address
+	 */
+	inline IPv4Address makeIPv4Address(uint32_t addrAsInt) { return IPv4Address(addrAsInt); }
+
+	/**
+	 * A function that creates an instance of the class IPv4Address out of 4-byte array.
+	 * @param[in] bytes The address as 4-byte array in network byte order
+	 * @return An instance of class IPv4Address
+	 */
+	inline IPv4Address makeIPv4Address(const uint8_t bytes[4]) { return IPv4Address(bytes); }
```

Comment:
These functions are included in N4771 and I followed to this document. We can remove them.

Reply:
These functions have been removed

Reply:
I usually don't use methods outside of classes, unless there is no class to attach them to. In this case since there is a c'tor already, I think they're redundant